nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py

```diff
@@ -0,0 +1,684 @@
+    """
+
+    def __init__(self, *args, max_width: int, node_name: str, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._active_width = max_width
+        self._max_width = max_width
+        self._node_name = node_name
+
+    @property
+    def max_width(self) -> width_type:
+        return self._max_width
+
+    def get_active_width(self) -> width_type:
+        return self._active_width
+
+    def set_active_width(self, width: width_type):
+        """
+        Set new number of filters that was set for preceding layers (input width) or for current layer
+        (output width). The operation will use this value to adapt parameters for the layer, to which the operation is
+        applied. The width should be less the original number of filters and more than one. Zero number of filters is
+        supported through Dynamic Depth feature.
+
+        :param width: number of filters that was set for preceding layers.
+        """
+        if width is None or width > self._max_width or width < 1:
+            raise AttributeError('Invalid width={} in scope={}.\nIt should be within the range: [1, {}]'.format(
+                width, self._node_name, self._max_width))
+
+        self._active_width = width
+
+
+class ElasticWidthOp(DynamicWidthOp):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._width_list = []
+
+    @property
+    def width_list(self) -> List[int]:
+        return self._width_list
+
+
+class ElasticWidthInfo(PrunedLayerInfoBase):
+    def __init__(self,
+                 node_name: NNCFNodeName,
+                 module: nn.Module,
+                 elastic_op: ElasticWidthOp,
+                 node_id: int,
+                 is_depthwise: bool):
+        super().__init__(node_name, node_id, is_depthwise)
+        self.module = module
+        self.elastic_op = elastic_op
+
+
+class DynamicLinearInputOp(DynamicWidthOp, nn.Module):
+    def forward(self, weight):
+        return weight[:, :self._active_width]
+
+
+class ElasticLinearOp(DynamicWidthOp, nn.Module):
+    def forward(self, weight, bias):
+        new_bias = None if bias is None else bias[:self._active_width].contiguous()
+        return weight[:self._active_width, :].contiguous(), new_bias
+
+
+class DynamicConvInputOp(DynamicWidthOp, nn.Module):
+    def forward(self, weight):
+        return weight[:, :self._active_width, :, :]
+
+
+class DynamicDWConvInputOp(DynamicWidthOp, nn.Module):
+    def forward(self, _):
+        return self._active_width
+
+
+class DynamicBatchNormInputOp(DynamicWidthOp, nn.Module):
+    SET_RUNNING_STATISTICS = False
+
+    def forward(self, **bn_params):
+        return [param[:self._active_width] for param in bn_params.values()]
+
+
+class ElasticWidthConv2DOp(ElasticWidthOp, nn.Module):
+    def __init__(self, max_out_channels, node_name, elastic_width_params: Dict[str, Any] = None):
+        super().__init__(max_width=max_out_channels, node_name=node_name)
+
```

Comment:
Could you please clarify what is the difference between `ElasticWidthConv2DOp` and `DynamicConvInputOp`?

Reply:
The first is for trimming output channels (elasticity), the latter - for input ones (adaptation to elasticity - dynamic trimming of input channels).

Will add more documentation to explain it. Thanks for noticing!

Reply:
@daniil-lyakhov renamed classes and added more docstrings. please review
```
ElasticWidthConv2DOp -> ElasticOutputWidthConv2DOp 
DynamicConvInputOp -> ElasticInputWidthConv2DOp 
```

Reply:
Now it's clear, thanks!